tests/common/dualtor/dualtor_utils.py

```diff
@@ -31,25 +31,56 @@ def mux_server_url(request, tbinfo):
     port = utilities.get_group_visible_vars(inv_files, server, 'mux_simulator_port')
     return "http://{}:{}/mux/{}".format(ip, port, vmset_name)
 
-def _url(server_url, physical_port, action):
+def _url(server_url, physical_port=None, action=None):
     """
     Helper function to build an url for given port and target
 
     Args:
         server_url: a str, the url for mux server, like http://10.0.0.64:8080/mux/vms17-8
         physical_port: physical port on switch, an integer starting from 1
-        action: a str, either "output" or "drop"
+                        If physical_port is none, the returned url contains no '/port/action' (For polling/toggling all ports)
+        action: a str, output/drop or None. If action is None, the returned url contains no '/action'
     Returns:
         The url for posting flow update request, like http://10.0.0.64:8080/mux/vms17-8/1/drop(output)
     """
+    if not physical_port:
+        return server_url
+    if not action:
+        return server_url + "/{}".format(physical_port - 1)
     return server_url + "/{}/{}".format(physical_port - 1, action)
 
+def _get(server_url):
+    """
+    Helper function for polling status from y_cable server.
+
+    Args:
+        server_url: a str, the full address of mux server, like http://10.0.0.64:8080/mux/vms17-8/[1]
+    Returns:
+        dict: A dict decoded from server's response.
+        None: Returns None is error is detected.
+    """
+    req = urllib2.Request(url=server_url)
+    try:
+        res = urllib2.urlopen(req)
+        data = res.read()
+        return json.loads(data)
+    except urllib2.HTTPError as e:
+        err_msg = json.loads(e.read().decode())['err_msg']
+        logger.warn("get request returns err. status_code = {} err_msg = {}".format(e.code, err_msg))
+    except urllib2.URLError as e:
+        logger.warn("get request returns err. err_msg = {}".format(str(e)))
+    except json.decoder.JSONDecodeError as e:
+        logger.warn("failed to parse response as json. err_msg = {}".format(str(e)))
+    except Exception as e:
+        logger.warn("get request returns err. err_msg = {}".format(str(e)))
+    return None
+
 def _post(server_url, data):
     """
     Helper function for posting data to y_cable server.
 
     Args:
-        server_url: a str, the full address of mux server, like http://10.0.0.64:8080/mux/vms17-8/1/drop(output)
+        server_url: a str, the full address of mux server, like http://10.0.0.64:8080/mux/vms17-8/[1/drop(output)]
```

Comment:
I think the convention is to use "|" , for example "this|that", to indicate either "this" or "that".